universal-application-tool-0.0.1/app/services/question/AddressQuestionDefinition.java

```diff
@@ -0,0 +1,31 @@
+package services.question;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import java.util.Locale;
+
+public class AddressQuestionDefinition extends QuestionDefinition {
+
+  public AddressQuestionDefinition(
+      String id,
+      String version,
+      String name,
+      String path,
+      String description,
+      ImmutableMap<Locale, String> questionText,
+      ImmutableMap<Locale, String> questionHelpText,
+      ImmutableSet<String> tags) {
+    super(id, version, name, path, description, questionText, questionHelpText, tags);
+  }
+
+  @Override
+  public QuestionType getQuestionType() {
+    return QuestionType.ADDRESS;
+  }
+
+  @Override
+  public ImmutableMap<String, Class> getScalars() {
+    return ImmutableMap.of(
+        "street", String.class, "city", String.class, "state", String.class, "zip", String.class);
```

Comment:
Issue (blocking): Java switch statements don't allow instances of `Class`, i.e. the following code doesn't compile:

```java
switch(questionDef.getScalars().get("street")) {
  case String.class: {
    ...
    break;
  }
}
```

Suggestion: another option could be to use an enum for the scalar types.

Reply:
It has been 0 days since I've been disappointed in Java.

Reply:
I considered the enum approach as well. I like it since we only want to support Java primitives and not any Class (at least for now)